### PR TITLE
github actions; all projects

### PR DESCRIPTION
**Pull request by jonapich**
This enables tests and publish on all projects.

There's a really weird behavior on mac where `platform.node()` changes from a `.` to a `-` mid execution. 🤷🏻‍♂️ 

Other than that, even though docker is installed on github action machines, they're configured for non-linux containers. The docker tests will fail trying to build the dockerfile. So I skipped those tests unless we're on linux.

The readmes were created just for the sake of the pyproject.toml and will be updated through #10

### CODE REVIEW

File: coveo-testing/tests_testing/test_test_id.py [hunk 1]
```diff
@@ -11,6 +11,8 @@
 from coveo_testing.parametrize import parametrize
 from coveo_testing.temporary_resource.unique_id import TestId, unique_test_id
 
+_ = unique_test_id  # mark fixtures are used
```
Comment (klalumiere) [inline coveo-testing/tests_testing/test_test_id.py:14]: Did you mean

```suggestion
_ = unique_test_id  # mark fixtures as used

```
?
(I feel like an auto-correct :laughing:)

File: .github/workflows/coveo-itertools.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+
+      - name: Publish to pypi
+        uses: ./.github/workflows/actions/publish-to-pypi
+        with:
+          project-name: ${{ github.workflow }}
+          pypi-token: ${{ secrets.PYPI_TOKEN }}
+          dry-run: ${{ github.ref != 'refs/heads/main' }}
```
Comment (klalumiere) [inline .github/workflows/coveo-itertools.yml:52]: I'm starting to guess that I'll review the same workflow with different `name` and `paths`. Perhaps you can write a local composite-action with parameterized `name`? I think you'd reap the benefit after you decide to update the workflows 2-3 times (for instance to update `python-version`).

**Review by klalumiere — CHANGES_REQUESTED** [review]
I'm quite uncomfortable with the 8 copy pastes. A single update will take 9 times longer to write and will also be longer to review. Moreover, if you add more packages, like `coveo-magic`, well that's a copy-paste more. We should at least discuss the possibility to use a local action with parameterized name. A typical counterargument against this approach is

>Yeah, but what if I want to make `.github/workflows/coveo-systools.hml` different?

and the counterargument to that counterargument is that, of course, you can. Once the local action is written, you still have the option to use it _or not_.

**Comment by jonapich** [comment]
> I'm quite uncomfortable with the 8 copy pastes. A single update will take 9 times longer to write and will also be longer to review. Moreover, if you add more packages, like `coveo-magic`, well that's a copy-paste more. We should at least discuss the possibility to use a local action with parameterized name. A typical counterargument against this approach is
> 
> >Yeah, but what if I want to make `.github/workflows/coveo-systools.hml` different?
> 
> and the counterargument to that counterargument is that, of course, you can. Once the local action is written, you still have the option to use it _or not_.

Originally I wanted to do composites but it was more problems than not. Is the "local action" à composite? For instance I couldn't call uses: python setup from a composite.

**Comment by klalumiere** [comment]
> > I'm quite uncomfortable with the 8 copy pastes. A single update will take 9 times longer to write and will also be longer to review. Moreover, if you add more packages, like `coveo-magic`, well that's a copy-paste more. We should at least discuss the possibility to use a local action with parameterized name. A typical counterargument against this approach is
> > > Yeah, but what if I want to make `.github/workflows/coveo-systools.hml` different?
> > 
> > 
> > and the counterargument to that counterargument is that, of course, you can. Once the local action is written, you still have the option to use it _or not_.
> 
> Originally I wanted to do composites but it was more problems than not. Is the "local action" à composite? For instance I couldn't call uses: python setup from a composite.

C'est dommage ça. Il y a déjà un [issue](https://github.com/actions/runner/issues/438) d'ouvert qui demande qu'on puisse faire ça si tu veux faire +1 :-).

**Review by klalumiere — APPROVED** [review]

**Pull request merged**